src/lib/storage.js

```diff
@@ -11,9 +11,18 @@ class Storage extends ScratchStorage {
         super();
         this.addWebSource(
             [this.AssetType.Project],
-            projectAsset => `${PROJECT_HOST}/${projectAsset.assetId}`
+            this.getProjectGetConfig.bind(this)
         );
     }
+    setProjectToken (projectToken) {
+        this.projectToken = projectToken;
+    }
+    getProjectGetConfig (projectAsset) {
+        const path = `${PROJECT_HOST}/${projectAsset.assetId}`;
+        const qs = this.projectToken ? `?token=${this.projectToken}` : '';
```

Comment:
why urlsearchparams? why not a header?

Reply:
I believe that is how this works on Fastly.